src/process.rs

```diff
@@ -183,30 +179,43 @@ pub async fn kinesis_logs(
         .clone()
         .unwrap_or_else(|| "NO SUBSYSTEM NAME".to_string());
     let v = &kinesis_message.0;
-    let string_data: Vec<u8> = if is_gzipped(v) {
-        // It looks like gzip, attempt to ungzip
-        match ungzip(v.clone(), String::new()) {
-            Ok(un_v) => un_v,
-            Err(_) => {
-                tracing::error!("Data does not appear to be valid gzip format. Treating as UTF-8");
-                v.clone()
-            }
-        }
+    let mut batches = Vec::new();
+    if config.integration_type == IntegrationType::CloudWatch {
+        tracing::debug!("CloudWatch IntegrationType Detected");
+        
+        let cloudwatch_payload = ungzip(v.clone(), String::new())?;
+        tracing::debug!("CloudWatch Payload {:?}", cloudwatch_payload);
```

Comment:
If we want to keep this debug log, I suspect logging `string_cw` would make more sense.